### PR TITLE
gh-87092: compiler's codegen stage uses int jump target labels, and the target pointer is only calculated just before optimization stage

### PR DESCRIPTION
**Pull request by iritkatriel**
* Issue: gh-87092

### CODE REVIEW

**Comment by iritkatriel** [comment]
Skipping news because this continues the work described in the news entry on #95398.

**Comment by iritkatriel** [comment]
The FreeBsd timeout seems unrelated - it's doing that on other PRs. 

This is ready to be reviewed now. With this I can write the unit tests without needing the test harness to worry about blocks.

File: Python/compile.c [hunk 1]
```diff
@@ -141,33 +144,31 @@ static struct location NO_LOCATION = {-1, -1, -1, -1};
...
 
-static struct jump_target_label_ NO_LABEL = {-1, NULL};
+static struct jump_target_label_ NO_LABEL = {-1};
 
 #define SAME_LABEL(L1, L2) ((L1).id == (L2).id)
 #define IS_LABEL(L) (!SAME_LABEL((L), (NO_LABEL)))
 
 #define NEW_JUMP_TARGET_LABEL(C, NAME) \
-    jump_target_label NAME = {cfg_new_label_id(CFG_BUILDER(C)), cfg_builder_new_block(CFG_BUILDER(C))}; \
+    jump_target_label NAME = {cfg_new_label_id(CFG_BUILDER(C))}; \
```
Comment (markshannon) [inline Python/compile.c]: Would it make sense to change `int cfg_new_label_id()` to `jump_target_label cfg_new_label()`?

**Review by markshannon — COMMENTED** [review]
One small suggestion, otherwise LGTM.

**Review by markshannon — APPROVED** [review]
:+1:

**Pull request merged**